op-batcher/batcher/channel.go

```diff
@@ -85,11 +85,18 @@ func (c *channel) TxConfirmed(id string, inclusionBlock eth.BlockID) bool {
 		c.log.Info("Channel is fully submitted", "id", c.ID(), "min_inclusion_block", c.minInclusionBlock, "max_inclusion_block", c.maxInclusionBlock)
 	}
 
-	// If this channel timed out, put the pending blocks back into the local saved blocks
-	// and then reset this state so it can try to build a new channel.
+	// If this channel timed out
+	// return true so that the the caller can reset the state and build a new channel.
 	if c.isTimedOut() {
 		c.metr.RecordChannelTimedOut(c.ID())
-		c.log.Warn("Channel timed out", "id", c.ID(), "min_inclusion_block", c.minInclusionBlock, "max_inclusion_block", c.maxInclusionBlock)
+		c.log.Warn("Channel timed out, invalidating", "id", c.ID(), "min_inclusion_block", c.minInclusionBlock, "max_inclusion_block", c.maxInclusionBlock)
+		return true
+	}
+
+	// If this channel had confirmed transactions straddling the Holocene activation time
+	// return true so that the the caller can reset the state and build a new channel.
+	if c.straddlesActivation(holoceneTime) {
```

Comment:
The expected progress check will only throw an error if there is a _fully submitted_ channel which implies more progress should be made (current l1 advanced beyond max inclusion block, safe l2 head behind latest l2 block of that channel).

At the moment a channel becomes fully submitted, if it is invalid (for any reason, timing out or straddling holocene activation), it will be immediately and synchronously deleted by the `channelManager.handleChannelInvalidated()` method. So the expected progress check would never range over such an invalid channel.